### PR TITLE
Update dependency for Symfony 4.

### PR DESCRIPTION
**Pull request by dbrumann**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | ---
| Documentation   | ---
| License         | MIT

#### What's in this PR?

Update dependency for symfony/stopwatch to allow 4.x as well to make this usable in Symfony 4-applications.

### CODE REVIEW

**Comment by joelwurtz** [comment]
Can you add also Symfony 4 to the build matrix of Travis ? (so we are sure there is no problem with it)

**Comment by dbrumann** [comment]
It looks like updating the build matrix doesn't work yet

**Comment by joelwurtz** [comment]
Symfony 4.0 is only in beta so that's normal, it will be released in november, so let's wait for the release to be available before pushing this.

File: .travis.yml [hunk 1]
```diff
@@ -30,6 +31,8 @@ matrix:
     include:
         - php: 5.4
           env: COMPOSER_FLAGS="--prefer-stable --prefer-lowest" COVERAGE=true TEST_COMMAND="composer test-ci" SYMFONY_VERSION=2.7.*
+        - php: 7.1
+          env: SYMFONY_VERSION=4.0.*
```
Comment (xabbuh) [inline .travis.yml]: should be `env: SYMFONY_VERSION=4.0.*@dev` since there is no stable release yet

**Comment by xabbuh** [comment]
@joelwurtz IMO allowing Symfony 4 now would be better as it allows to catch incompatibilities before the final release. People not relying on dev dependencies would still not get 4.0 components, but would have to opt-in explicitly if they prefer to use it now.

File: .travis.yml (hunk 1 above)
Comment (dbrumann) [reply .travis.yml]: Yes, but this will still break, because phpspec will complain: https://github.com/phpspec/phpspec/blob/master/composer.json#L29-L33

**Comment by dbrumann** [comment]
@joelwurtz I think adding Symfony 4 to the build matrix does not make sense as we will have to exclude everything except PHP 7.1 anyways as they are not supported by Symfony 4.

**Comment by dbu** [comment]
we don't want every symfony version with every php version, that explodes the build matrix. i think this looks right for the configuration.

do you see where the installation conflict comes from?

**Comment by dbrumann** [comment]
@dbu It boils down to phpspec. When requiring symfony/symfony it conflicts with symfony/stopwatch which can not be updated due to symfony/yaml which can not be updated due to phpspec which (as far as I can tell) can not be updated due to the coverage plugin and the version restriction in the composer.json (current version is 4.2.4).

I am currently trying to make this work on my local machine without breaking the old requirements, but I'm having a hard time. Suggestions are very welcome :)

**Comment by Nyholm** [comment]
Have a look at https://github.com/php-http/client-common/pull/90. I want to make all travis files like that.

File: .travis.yml (hunk 1 above)
Comment (dbu) [reply .travis.yml]: i think we can use COMPOSER_FLAGS="--minimum-stability=dev" or something like that.

File: .travis.yml [hunk 2]
```diff
@@ -29,19 +22,33 @@ matrix:
     fast_finish: true
     include:
         - php: 5.4
-          env: COMPOSER_FLAGS="--prefer-stable --prefer-lowest" COVERAGE=true TEST_COMMAND="composer test-ci" SYMFONY_VERSION=2.7.*
+          env: DEPENDENCIES=minimum COVERAGE=true TEST_COMMAND="composer test-ci" SYMFONY_VERSION=2.7.*
+        - php: 5.5
+        - php: 5.6
+        - php: 7.0
+        - php: 7.1
+          env: SYMFONY_VERSION=3.0.*
```
Comment (xabbuh) [inline .travis.yml]: Why `3.0.*`? Didn't you mean `3.4.*` instead?

File: .travis.yml [hunk 3]
```diff
@@ -29,19 +22,33 @@ matrix:
...
+        - php: 7.1
+          env: DEPENDENCIES="dev" SYMFONY_VERSION=4.0.*
+        - php: hhvm
     exclude:
         - php: 5.4
           env: SYMFONY_VERSION=3.0.*
```
Comment (xabbuh) [inline .travis.yml]: This exclude role does not match any entry from the matrix above.

File: .travis.yml (hunk 2 above)
Comment (dbrumann) [reply .travis.yml]: I kept this requirements from the selection above. Something like `env: DEPENDENCIES="symfony/lts:^3"` probably makes more sense?

Comment (xabbuh) [reply .travis.yml]: I would also update the other places where `symfony/symfony` is used by using `symfony/lts` instead. That way you won't hide bugs related to not installed components outside the CI environment.

Comment (dbu) [reply .travis.yml]: i guess best see https://github.com/symfony/symfony-docs/pull/8701/ - or lets just make things work here and then sync to that when the recommendation for a perfect .travis.yml has been merged.

**Review by dbu — COMMENTED** [review]
thanks. (and sorry for my comment on the partial diff, seems this has meanwhile been improved anyways)

looking at travis, the symfony 4.0 build does not work (but is allowed to fail) because phpspec can not be installed. only master of phpspec (4.2.x) does allow symfony 4: https://github.com/phpspec/phpspec/blob/master/composer.json . but unfortunately its limited to symfony 3.2 or newer, so we can't just upgrade to that. 

maybe we could upgrade to phpspec 3 which does support sf 2.7 to 3.* and add something to the sf 4 build to go to that phpspec version. of course i don't know if there are any bc breaks between 3 and 4 that would affect us.

**Comment by fbourigault** [comment]
If we remove `henrikbjorn/phpspec-code-coverage` and use `phpspec/phpspec` `^2.5 || ^3.0 || ^4.0` specs are greens. So having `henrikbjorn/phpspec-code-coverage` compatible with `phpspec/phpspec` `^4.0` will probably solve the issue. The other possibility is to drop code coverage ^^.

**Comment by dbu** [comment]
i suggest we comment out the coverage thing for the moment and create an issue to reenable coverage when the henrikbjorn tool is ready for phpspec 4.

**Comment by xabbuh** [comment]
You could collect the code coverage report on a job that doesn't use Symfony 4 only.

**Comment by dbu** [comment]
or that. but would need an extra composer require for the henrikbjorn library. if its in composer.json, composer can't install in the symfony 4 test.

**Comment by Nyholm** [comment]
This is save to merge. Looking at the diff in the [symfony component](https://github.com/symfony/stopwatch/compare/v3.3.13...master), you will see that there is no BC breaking change. 
I suggest merging this. I will fix our broken tests for dev now. (different PR)

**Comment by Nyholm** [comment]
@fbourigault told me not to "fix" the tests. He told me to fix the PR according to the suggestions.

**Review by fbourigault — APPROVED** [review]

**Comment by Nyholm** [comment]
Thank you Denis!

**Pull request merged**